### ISSUE TITLE
Stalker Anomaly 1.5.0 update 3 glitched text

### ISSUE THREAD

**Issue by wychuzord** [doitsujin/dxvk#1246]
Some text gets glitched in following fashion:
https://files.catbox.moe/hnmab4.png
https://files.catbox.moe/ndjt29.png
https://files.catbox.moe/nkc632.png

When using these https://drive.google.com/drive/folders/1-gcaFB0nazU-ucUy0wycxIVmIL72Ogqs dlls inside Anomaly's /bin/ folder the problem was not happening.

### Software information
Anomaly 1.5.0 beta update 3

### System information
- GPU: RX 5700 XT
- Driver: DRM 3.35.0, 5.4.0-050400rc5-generic, LLVM 9.0.0, Mesa 20.0.0-devel (git-01cae57 2019-11-12 eoan-oibaf-ppa)
- Wine version: 4.19 staging
- DXVK version: 1.4.4

### Apitrace file(s)
https://files.catbox.moe/or9ker.rdc
Renderdoc file. I was advised by someone to use renderdoc for Vulkan because Apitrace on linux is only for OpenGL. If it's not sufficient I'll try to wrestle with Apitrace. Unable to capture frame after loading or starting new game because I'm getting ring gfx_0.0.0 error each time (driver issue unrelated to DXVK).

### Log files
- d3d11.log: https://files.catbox.moe/z74qhb.log
- dxgi.log: https://files.catbox.moe/fm6rz4.log

**Comment by K0bin** [comment]
>  Renderdoc file. I was advised by someone to use renderdoc for Vulkan because Apitrace on linux is only for OpenGL. If it's not sufficient I'll try to wrestle with Apitrace. Unable to capture frame after loading or starting new game because I'm getting ring gfx_0.0.0 error each time (driver issue unrelated to DXVK).

You have to use the Windows version with Wine. More info here: https://github.com/Joshua-Ashton/d9vk/wiki/Making-a-Trace (I know it's a guide for D9VK but the steps are mostly the same except that you'll have to use d3d11.dll, dxgi.dll and dxgitrace.dll instead of d3d9.dll)

**Comment by doitsujin** [comment]
Renderdoc captures **only** work on the **same** GPU with the **same** driver, so sharing those is pretty much useless. I can't open it on my RX 480.

> When using these https://drive.google.com/drive/folders/1-gcaFB0nazU-ucUy0wycxIVmIL72Ogqs dlls inside Anomaly's /bin/ folder the problem was not happening.

And what **are** those DLLs? If that's an older DXVK build, please just mention the version.

**Comment by doitsujin** [comment]
Works fine for me:
![Bildschirmfoto-1058](https://user-images.githubusercontent.com/25567304/68746390-c7c7a680-05f8-11ea-997a-e91362647359.png)

**Comment by wychuzord** [comment]
The one that has glitchless text is DXVK v1.0-7-gd011102. Author of that guide I linked told that he didn't exactly know what they are or where did he get them from so hell if I know.
Here's proof https://files.catbox.moe/pqs41s.png
Tried running it it a 1920x1080 and it's the same. I'll try getting the apitrace. It seems like Cyrillic script is interfering with English. It oddly looks similar to when icons for items inside inventory are misaligned but it was the same before I added any submods (and I only have HD models anyway)...

**Comment by wychuzord** [comment]
When I tried d3d11.dll from apitrace I couldn't run the game with library override but without it (or any library overrides for that matter) it did run, it didn't give me any problems with text but...
https://files.catbox.moe/di9tux.png (A-Life becoming self-aware lol)
Also this "DO NOT use DXVK together with apitrace!" will pose a problem. I don't think game will run in dx11 at all without DXVK. Before I found a way to do it with DXVK I could only get dx9 executable to work.

**Comment by K0bin** [comment]
Just d3d11.dll isn't enough for apitrace. You need dxgi.dll and dxgitrace.dll too.
https://github.com/doitsujin/dxvk/wiki/Common-issues#apitrace

**Comment by wychuzord** [comment]
TL;DR: I REALLY DON'T THINK I CAN RUN IT IN DX11 WITHOUT DXVK
And more importantly even if I could the problem is triggered by something inside DXVK 1.4.4 libraries anyway so I don't know how would I trace it with apitrace libraries.

In a prefix that had DXVK on it with these 3 dlls from apitrace in /Anomaly/bin/ it's just the "terror not a morning" situation. With these 3 ddls and library overrides I get an error I didn't see before:
https://files.catbox.moe/xq8glq.png
With clean prefix without dll overrides I get standard "install dx11 you twat" prompt (https://files.catbox.moe/z7jzo4.png), with overrides I get yet another bug I didn't see:
https://files.catbox.moe/qg9t52.jpg 
And, for the curious, the log reads:
* Detected CPU: AMD Ryzen 5 3600 6-Core Processor               [AuthenticAMD], F15/M1/S0, 4199.00 mhz, 37-clk 'rdtsc'
* CPU features: RDTSC, MMX, SSE, SSE2, SSE3, SSSE3, SSE4.1, SSE4.2, HTT
* CPU cores/threads: 6/12

Found commandline file!
Initializing File System...
using fs-ltx fsgame.ltx
File System Ready...
FS: 61333 files cached 47 archives, 992Kb memory used.
Init FileSystem 0.987728 sec
'xrCore' build 7541, Sep 29 2019

EH: 3CAABCFCFF6F3A810019C6A72180F166

-----loading c:/path/to/the/game/anomaly\gamedata\configs\system.ltx
-----loading c:/path/to/the/game/anomaly\gamedata\configs\system.ltx
Initializing Engine...
Starting INPUT device...
Loading DLL: xrRender_R4.dll
stack trace:

 
[error][       2] : File not found.

**Comment by K0bin** [comment]
Those errors mention d3d**x**11. That's a helper library that interacts with d3d11. Make sure you're using the Microsoft version of that as the Wine one is broken. You can install that with Winetricks/Protontricks.

**Comment by wychuzord** [comment]
Could you point me to an exact d3d11 in Winetricks? Not sure which is which. I think I'm using d3d11_43.dll and as long as DXVK is used with it it works fine (aside from text errors). I get this error when I try to run Anomaly with DXVK but without d3d11 or d3dcompiler.

**Comment by sputniktr02** [comment]
I am currently playing Anomaly and I don't have this issue; for reference these are what I have installed with winetricks:

d3dcompiler_42
d3dcompiler_43
d3dcompiler_47

d3dx10_43
d3dx10
d3dx11_42
d3dx11_43

I am on an Nvidia GTX860m

**Comment by wychuzord** [comment]
Just tried it in new prefix, just to be sure. Nothing changed. Without DXVK I got the "terror not a morning" situation, with DXVK (1.4.5) I had borked text.
I'm assuming it is something Navi 10 related but it's not just Navi 10 issue because with those dlls from the google folder I linked (1.0-7-gd011102) I was able to run it without the glitched text so the newer DXVK does something wrong.

**Comment by doitsujin** [comment]
So what **exactly** did you do to reproduce the problem? What **exactly** are your in-game settings?

**Comment by wychuzord** [comment]
Steps:
1. Using winetricks I create new 64 bit prefix, I choose "install a Windows  dll or component" and pick: d3dcompiler (tested with just _43 or _47 - no difference) 
d3dx11_43 (don't remember if I tried 42 alone but when I installed every dll from post above there was no change)
dxvk
(normally I'd also add quartz for in-game movies but it didn't affect the problem)
3. Run explorer, choose AnomalyLaucher.exe
4. Choose DirectX 11 (R4), Shadow Map 2048 (all options same problem), everything else ticked off.

As for in-game settings I assume only Video tab is relevant:
https://files.catbox.moe/yfoni2.png
https://files.catbox.moe/ej7gdy.png
https://files.catbox.moe/if4gy4.png
https://files.catbox.moe/tpfuwg.png
https://files.catbox.moe/1zs0x4.png

**Comment by doitsujin** [comment]
So i followed your instructions but I **still** cannot reproduce the problem here:
![Bildschirmfoto-1076](https://user-images.githubusercontent.com/25567304/69460302-151fe280-0d74-11ea-8f96-886c74c496d2.png)

Not sure what's going on but DXVK seems to be doing fine.

**Comment by wychuzord** [comment]
But you're using RX 480 right? As I said it's highly probable the problem is caused by something in Navi 10 but then again with the other dlls the problem simply wasn't there. I don't get it.
I understand I haven't given you any data you could work with, I just hoped maybe you'd miraculously remember or notice what was the difference between these dlls and current DXVK dlls affecting the text. 
The game runs like ass anyway (I know, because decompiler being employed) even with newer DXVK so I might as well use those old dlls.

**Comment by wychuzord** [comment]
I've made a discovery.
Fonts work properly with dxvk 1.3.3/1.3.4 available through lutris but will bork out with anything higher than this. 
There is something in 1.4 that causes problems with Navi 10 with oibaf drivers and it isn't only this game. Recently I had Nier: Automata start displaying weird colours in cracking sequences (like the normally gray map being yellow, reddish or green tinted) and missing shaders in Bunker area (lift leading to the command bridge being bright gray in contrast to dark gray all around it, npc clothes being brighter than they should be). It all worked nicely with 1.4.5 couple days ago but one of the recent patches caused every dxvk version past 1.4 to lack some shaders or display weird colours in cracking sequences. Importantly 1.3.3 and 1.3.4 don't have these problems.

**Comment by wychuzord** [comment]
Actually I was wrong about Nier Automata - every dxvk version causes some problems, it's just the tint difference in 1.3.3 and 1.3.4 was harder to notice. It's somewhat random too, I managed somehow to get 1.4.4 to work properly once and then it wen't back to heavy red backgrround.
Point being what's happening with Nier Automata is unrelated to what's happening with Anomaly.

**Comment by wychuzord** [comment]
Problem has been solved for me. I thought it was DXVK 1.5.1 but it works on older versions now as well. Runner version also didn't change anything in that regard. Something in recent drivers must've fixed it.
